src/protocol/templates/ping_protocol.json

```diff
@@ -69,661 +69,469 @@
     ],
     "messages": {
         "ping1D": {
-            "undefined": {
-                "id": "0",
-                "payload": []
+            "general": {
+                "undefined": {
+                    "id": "0",
+                    "payload": []
+                },
+                "ack": {
+                    "id": "1",
+                    "description": "Acknowledged.",
+                    "payload": []
+                },
+                "nack": {
+                    "id": "2",
+                    "description": "Not acknowledged.",
+                    "payload": [
+                        {
+                            "name": "id_to_nack",
+                            "type": "u16",
+                            "description": "The message ID being NACKed."
+                        },
+                        {
+                            "name": "err_msg",
+                            "type": "char[var]",
+                            "description": " NULL terminated ASCII text message."
+                        }
+                    ]
+                },
+                "ascii_text": {
+                    "id": "3",
+                    "description": "A message for transmitting text data.",
+                    "payload": [
+                        {
+                            "name": "msg",
+                            "type": "char[var]",
+                            "description": " NULL terminated ASCII text message."
+                        }
+                    ]
+                }
             },
-            "ack": {
-                "id": "1",
-                "description": "Acknowledged.",
-                "payload": []
+            "set": {
+                "set_device_id": {
```

Comment:
the "set_" prefix is not necessary here, the parent already specify this messages as "set".

Reply:
yes, but I think that taking it out will require more code to do what we need == our jobs harder

also the messages should be keyed by their name, if we do it the other way then we have messages with the same key.

Reply:
I'm open to refactoring it, right now I have some code around the current patch

Reply:
I thought that you was aiming to have get_foo() and set_foo(), but now I see that you are using it to create foo() and set_foo(), is that correct ?

Reply:
correct.